### DISCUSSION TITLE
生成背景水印图失败

### DISCUSSION THREAD

**Discussion by lanyeeee** [lanyeeee/bilibili-manga-watermark-remover#1] [General]
|颜色1(black.png)|颜色2(white.png)|
|-|-|
|![image](https://github.com/user-attachments/assets/927088ff-c130-40f7-b985-459162c2dcb9) |![image](https://github.com/user-attachments/assets/a5cc8b1e-b57f-4fd8-8b5a-d844da045dca) |

要**成功**生成背景水印图，需要找到两张`水印附近是纯色`且`纯色差值>50`的图片。例如上面这两张图
而如果相应尺寸的图片里没有两张`水印附近是纯色`且`纯色差值>50`的图片，就会提示生成失败
目前并没有想到什么太好的解决办法

下面是一张无法作为背景水印图的图片，因为水印附近不是纯色
![image](https://github.com/user-attachments/assets/b09b10e6-c8b3-48ed-8395-fb75ace350b6)

**Comment by Zzy-114514** [comment]
有些彩色漫画找不到黑色背景图片，所以可不可以通过其他颜色的背景求出水印？
原本知道out和in（0和255）求出watermark和alpha：out = in * alpha + watermark * (1 - alpha)
但在数学的角度，即使in不是0和255时也可以求解，而且实际情况彩色的纯色背景图肯定有多张，可以用来校准

**Reply by lanyeeee** [reply]
可以通过其他颜色的背景求

确实理论上black和white只要有色差就可以
但是如果差太小，肯定会因为浮点数的精度而出现问题
具体多少要多少色差，之后再编写代码验证

另外，彩漫的去水印效果并不好

**Comment by ssbroad** [comment]
是不是漫画里必须同时要有一张白色背景（这个好找），一张黑色背景（这个大部分漫画没有）

缺一个都无法启动？

**Reply by lanyeeee** [reply]
是，具体看[去水印原理](https://github.com/lanyeeee/bilibili-manga-watermark-remover?tab=readme-ov-file#%E5%8E%BB%E6%B0%B4%E5%8D%B0%E5%8E%9F%E7%90%86)
不过具体来说是要两张像素值之差>50的背景，不一定要黑+白，具体看[背景水印图生成逻辑](https://github.com/lanyeeee/bilibili-manga-watermark-remover/discussions/55)

**Reply by fwxb** [reply]
挺好用的，找到两张作为两个原型件就行，给作者点赞
